mypy/build.py

```diff
@@ -1838,7 +1844,7 @@ def __init__(self,
         if path and source is None and self.manager.fscache.isdir(path):
             source = ''
         self.source = source
-        if path and source is None and self.manager.cache_enabled:
+        if path and self.manager.cache_enabled:
```

Comment:
Could you explain what's going on here? Naively I'd expect a change on L1844

Reply:
Yeah, the idea was that we still need to parse the cache even when it is a namespace package (because it hit that `isdir` check just above). I just changed it to restore the old check, but sank the `isdir` stuff futher down